Plots/MapProjections/NCL_sat_1.py

```diff
@@ -0,0 +1,380 @@
+# Import packages:
+import xarray as xr
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import numpy as np
+from sklearn.cluster import DBSCAN
+import warnings
+
+import geocat.datafiles as gdf
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and
+# load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/slp.1963.nc"), decode_times=False)
+
+# Get data from the 24th timestep
+pressure = ds.slp[24, :, :]
+
+# Translate short values to float values
+pressure = pressure.astype('float64')
+
+# Convert Pa to hPa data
+pressure = pressure*0.01
+
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_pressure = gvutil.xr_add_cyclic_longitudes(pressure, "lon")
+
+###############################################################################
+
+
+def findLocalExtrema(da, highVal=1040, lowVal=975, eType='Low'):
+    """
+    Utility function to find local low/high field variable coordinates on a contour map. To classify as a local high, the data
+    point must be greater than highVal, and to classify as a local low, the data point must be less than lowVal.
+    Args:
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable (ex. pressure) data values
+        highVal (:class:`int`):
+            Data value that the local high must be greater than to qualify as a "local high" location.
+            Default highVal is a pressure value of 1040 hectopascals.
+        lowVal (:class:`int`):
+            Data value that the local low must be less than to qualify as a "local low" location.
+            Default lowVal is a pressure value of 975 hectopascals.
+        eType (:class:`str`):
+            'Low' or 'High'
+            Determines which extrema are being found- minimum or maximum, respectively.
+            Default eType is 'Low'.
+    Returns:
+        clusterExtremas (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify local low/high locations
+    """
+
+    # Create a 2D array of coordinates in the same shape as the field variable data
+    # so each coordinate is easily mappable to a data value
+    coordarr = []
+    for y in np.array(da.lat):
+        temparr = []
+        for x in np.array(da.lon):
+            temparr.append((x, y))
+        coordarr.append(temparr)
+    coordarr = np.array(coordarr)
+
+    # Set number that a derivative must be less than in order to
+    # classify as a "zero"
+    bound = 0.0
+
+    # Get global gradient of contour data
+    grad = np.gradient(da.data)
+
+    # Gradient in the x direction
+    arr1 = grad[0]
+
+    # Gradient in the y direction
+    arr2 = grad[1]
+
+    # Get all array 1 indexes where gradient value is between -bound and +bound
```

Comment:
Since bound is equal 0.0, do you need to be checking the interval [-bound .. bound]?

Reply:
Not when the bound is 0... I kept that in there just in case the user wanted to change the bound, but I will take it out.